vowpalwabbit/cb_explore_adf.cc

```diff
@@ -250,7 +250,8 @@ void predict_or_learn_first(cb_explore_adf& data, multi_learner& base, multi_ex&
   {
     float prob = 1.f / (float)num_actions;
     for (size_t i = 0; i < num_actions; i++) preds[i].score = prob;
-    data.tau--;
+    if (is_learn && data.gen_cs.known_cost.probability < 1 && test_adf_sequence(examples) != nullptr)
```

Comment:
Why the extra clauses in the 'if'?

Reply:
I copy pasted from line 241. I'm happy to remove them but I wanted to be in sync with how we determine if learn or predict

Reply:
Should the extra clauses be removed also from line 241?

Reply:
Likely yes.   "learn" should only be triggered when there is a valid label.

Reply:
`data.gen_cs.known_cost.probability < 1` should be removed since we are not checking this in any other exploration mode
`test_adf_sequence(examples) != nullptr` is checking if the sequence is valid. If we remove it then who is responsible to check if we have a valid label. Do you mean that whoever is calling `predict_or_learn_first` with is_learn=true must be responsible for the validity of the label?